include/envoy/filesystem/filesystem.h

```diff
@@ -61,5 +61,30 @@ class Watcher {
 
 typedef std::unique_ptr<Watcher> WatcherPtr;
 
+/**
+ * Abstraction for listing a directory.
+ * TODO(sesmith177): replace with std::filesystem::directory_iterator once we move to C++17
+ */
+
+class DirectoryIterator {
+public:
+  enum class FileType { Regular, Directory, Other };
+
+  typedef struct {
+    std::string path_;
```

Comment:
These fields need comments. E.g. is `path_` the full path or just the file name?